packages/debug/src/index.ts

```diff
@@ -61,30 +60,20 @@ const topProps = {
   },
   log: (...args: string[]) => {
     const [namespace, format, ...rest] = args
-    let logger: (...args: unknown[]) => void
-
-    if (
-      typeof require === 'function' &&
-      typeof process !== 'undefined' &&
-      typeof process.stderr !== 'undefined' &&
-      typeof process.stderr.write === 'function'
-    ) {
-      logger = (...args: unknown[]) => {
-        // On react native util is not defined but the entire checks above pass
-        // that's why we need to catch the error
-        try {
-          const util = require(`${'util'}`)
-          process.stderr.write(util.format(...args) + '\n')
-        } catch (e) {
-          logger = console.warn ?? console.log
-        }
+    const baseLogger = console.warn ?? console.log
+
+    const logWithFormatting = (() => {
+      try {
+        // TODO: try and see whether `require('node:util')` works in React Native.
+        const util = require(`${'util'}`)
+        return (...innerArgs: unknown[]) => baseLogger(`${util.format(...innerArgs)}`)
```

Comment:
I think you don't need to manually call `util.format` if you are using `console` methods